pkg/endpoint/endpoint.go

```diff
@@ -2335,3 +2343,12 @@ func (e *Endpoint) setDefaultPolicyConfig() {
 func (e *Endpoint) GetCreatedAt() time.Time {
 	return e.createdAt
 }
+
+// SetEndpointSyncControllerName sets the name of the endpoint controller used
+// to synchronize the endpoint itself to kubernetes.
+func (e *Endpoint) SetEndpointSyncControllerName(name string) {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+
+	e.endpointSyncControllerName = name
+}
```

Comment:
I have a suggestion that is more aesthetic than it is functional:

What if we instead move `controllerNameOf()` into the endpoint package rather than storing an extra string in the `Endpoint` struct? This way the ep syncher watcher can call `controllerNameOf()` and then when we manually trigger the controller, we call `controllerNameOf()` to get the name of the controller to trigger.

Reply:
Yep, makes sense :+1: I was initially a bit hesitant as I felt we were mixing responsibilities (since this function is about a _controller_ name) but in the end it also relates to endpoints so I think it's fine to move it in the `endpoint` package